### ISSUE TITLE
astro-fuse: Type `'{ basedOn: "output"; }'` is not assignable to type `'OutputBaseAstroFuseConfig'`

### ISSUE THREAD

**Issue by bennycode** [johnny-mh/blog2#18]
I tried this config in my `astro.config.ts` file:

```ts
fuse({
  basedOn: 'output',
})
``` 

But it wants me to define a `getFn` function as `Fuse.FuseIndexOptions` has `getFn` as a required property.

**Comment by johnny-mh** [comment]
Hi @bennycode

You can pass `getFn` directly `fuse` method.While it currently lacks type support (types depend on content), you can directly pass getFn to achieve the same functionality for creating the index file.

```ts
fuse({getFn: () => ... })
/**
 * INSIDE ASTRO-FUSE
 * using Fuse.createIndex(
 *   keys,
 *   {pathname: '', content: '...', frontmatter: {}}[],
 *   {getFn: () => ...} <-
 * )
 */
```

Unfortunately, due to a bug, you can't currently pass getFn within the keys option.

(I'll resolve this issue soon. but it may be hard because load astro.config and extract function and write it to html page)

```ts
    fuse({
      keys: [
        'content',
        'frontmatter.title',
        { name: 'tag', getFn: (obj) => obj },
      ],
    }),
```

![image](https://github.com/johnny-mh/blog2/assets/1061205/fe2529b4-84e2-4ce1-99ae-72c939fe15b5)

You can create a custom loadFuse function as an alternative solution. You can refer to the `injectScript` API for details on how to achieve this.

Thank you

**Comment by bennycode** [comment]
Hi @johnny-mh and thank you for your fast reply.

I actually don't want to supply a `getFn` method but as soon as I define `basedOn: 'output'` the TypeScript typings want to force me to supply a `getFn` method. 

**Screenshot:**

![image](https://github.com/johnny-mh/blog2/assets/469989/1ee2a5a7-d56f-4f12-8dc5-5c9d2a080786)

My current workaround is the following:

```ts
import fuse from 'astro-fuse';
import type { AstroFuseConfig } from 'astro-fuse/dist/types';

fuse({
  basedOn: 'output',
} as AstroFuseConfig
```  

I would like to avoid the type assertion `as AstroFuseConfig` if possible?

**Comment by johnny-mh** [comment]
I have just published version 0.0.7 with updated function argument types. Please check it out when you have a chance. If you have any issues, please feel free to ask. Thank you!

**Comment by bennycode** [comment]
**Version 0.0.7** works fine with my config! I was able to remove the `as AstroFuseConfig` type assertion. 🤘

Thanks for building `astro-fuse` and following up on my issues. I am applying your search functionality here: https://typescript.tv/search/ 😃
